src/Core/Grid/Column/Type/OrderReturnCustomizationColumn.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Column\Type;
+
+use PrestaShop\PrestaShop\Core\Grid\Column\AbstractColumn;
+
+/**
+ * Class OrderReturnColumn defines column that renders customization data for order return products
+ */
+class OrderReturnCustomizationColumn extends AbstractColumn
```

Comment:
Looks like this column is just a configured column, IIRC we have a configurable column that can be used to avoid creating files like this one.

ping @jolelievre @mflasquin

Reply:
@FabienPapet which configurable column are you refering to? As far as I know we don't have configurable columns 🤔

However, what you could do @JevgenijVisockij, since this column has no intelligence and no specific options you can use a basic `DataColumn` (with `field = customizations`) and then you can override the column template for this specific column. The grid system has an automatic template override mechanism based on:
- grid ID
- column ID
- column type

So in the column templates folder https://github.com/PrestaShop/PrestaShop/tree/8.1.x/src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Columns/Content You can add a `order_return_products_customizations_data.html.twig` that will replace the default twig template and you can handle your custom template here

Reply:
Also I believe what @FabienPapet suggested, but I don't think it exists yet, is to have a column in which you could simply specify a twig template to use. You can also create such a column which would come handy for modules and even for us because we can store our custom twig templates in more specified folders instead of increasing the grid default one forever.

Reply:
So you could introduce a `TemplateColumn` that extends `AbstractColumn` and adds a required option `template_path` so you can basically use ut this way:

```php
            ->add(
                (new TemplateColumn('customizations'))
                    ->setName($this->trans('Customizations', [], 'Admin.Global'))
                    ->setOptions([
                        'template_path' => '@PrestaShop/Admin/Sell/CustomerService/OrderReturn/customizations_column.html.twig',
                    ])
            )
```

I don't think it needs any more options for now, since inside it you can pretty much do any custom hard coded data retrieval, and if needed it can always be added later.

Reply:
Oh and in the generic layout `src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Columns/Content/custom_template.html.twig` you will only have to include the specified template path:

```twig
{{ include(column.options.template_path) }}
```
or
```twig
{{ include(column.options.template_path, {'record': record, 'column': column, 'grid': grid}) }}
```
because I don't know if the variables will be accessible out of the box or not, you need to try.

Reply:
I will go with TemplateColumn, IMO that makes the most sense

Reply:
That would make columns classe more reusable, thank you @jolelievre for the explanation. That's  what I was thinking.